### PR TITLE
feat/equalize3d

### PR DESCRIPTION
**Pull request by vribeiro1**
### Description
Implements equalize function for a sequence of images using PyTorch ops. Considers the histogram of the volume on the equalization calculations.

_Note for the reviewer: The original code for equalize function does not have tests._

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [x] Did you discuss the functionality or any breaking changes before ?
- [x] **Pass all tests**: did you test in local ? `make test`
- [x] Unittests: did you add tests for your new functionality ?
- [x] Documentations: did you build documentation ? `make build-docs`
- [x] Implementation: is your code well commented and follow conventions ? `make lint`
- [x] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

**Comment by shijianjian** [comment]
LGTM. Can you add ```kornia.augmentation.augmentation3d.RandomEqualize3D``` as well?

I only wrote the doctest for ```augmentation.py``` back to that time, please feel free to add new test. For running doctest, you may use ```python -m doctest kornia/augmentation/augmentation.py```

**Review by edgarriba — CHANGES_REQUESTED** [review]
unit test for new functionalities are mandatory too.

**Comment by edgarriba** [comment]
@shijianjian @vribeiro1 open a separate PR for `RandomEqualize3D`.
@shijianjian regarding doctest, could you add an automatic trigger for it ?

**Comment by vribeiro1** [comment]
@shijianjian @edgarriba I will write the appropriate tests and open a second PR for RandomEqualize3D. Thank you for your comments!

**Review by shijianjian — CHANGES_REQUESTED** [review]
LGTM. Please add more test.

File: test/enhance/test_adjust.py [hunk 1]
```diff
@@ -599,3 +599,150 @@ def test_gradcheck(self, device):
...
+        ])
+        expected = self.build_input(channels, height, width, bs=1, row=row_expected)
+        expected = expected.to(device)
+
+        f = kornia.enhance.equalize
+
+        assert_allclose(f(inputs), expected)
+
+    def test_equalize_batch(self, device):
+        bs, channels, depth, height, width = 2, 3, 6, 20, 20
```
Comment (shijianjian) [inline test/enhance/test_adjust.py:642]:
```suggestion
        bs, channels, height, width = 2, 3, 20, 20
```

File: test/enhance/test_adjust.py [hunk 2]
```diff
@@ -599,3 +599,150 @@ def test_gradcheck(self, device):
...
+        bs, channels, height, width = 2, 3, 4, 5
+
+        inputs = torch.ones(bs, channels, height, width)
+        inputs = inputs.to(device)
+        f = kornia.enhance.equalize
+
+        assert f(inputs).shape == torch.Size([bs, channels, height, width])
+
+    def test_equalize(self, device):
+        bs, channels, depth, height, width = 1, 3, 6, 20, 20
```
Comment (shijianjian) [inline test/enhance/test_adjust.py:624]:
```suggestion
        bs, channels, height, width = 1, 3, 20, 20
```

**Review by shijianjian — APPROVED** [review]

File: kornia/enhance/adjust.py [hunk 3]
```diff
@@ -430,46 +462,35 @@ def equalize(input: torch.Tensor) -> torch.Tensor:
...
     for image in input:
         # Assumes RGB for now.  Scales each channel independently
         # and then stacks the result.
-        scaled_image = torch.stack([scale_channel(image, i) for i in range(len(image))])
+        scaled_image = torch.stack([_scale_channel(image[i, :, :]) for i in range(len(image))])
         res.append(scaled_image)
     return torch.stack(res)
 
 
+def equalize3d(input: torch.Tensor):
```
Comment (edgarriba) [inline kornia/enhance/adjust.py:474]: missing return type in definition `def equalize3d(input: torch.Tensor) -> torch.Tensor:`

File: kornia/enhance/adjust.py [hunk 4]
```diff
@@ -430,46 +462,35 @@ def equalize(input: torch.Tensor) -> torch.Tensor:
...
+    """Implements Equalize function for a sequence of images using PyTorch ops based on uint8 format:
+    https://github.com/tensorflow/tpu/blob/master/models/official/efficientnet/autoaugment.py#L352
+    Args:
+        input (torch.Tensor): image tensor with shapes like (C, D, H, W) or (B, C, D, H, W) to equalize.
+    Returns:
+        torch.Tensor: Sharpened image or images.
+    """
```
Comment (edgarriba) [inline kornia/enhance/adjust.py:481]: docs won't render well - add this instead:
r"""Equalizes the values for a 3D volumetric tensor.

Implements Equalize function for a sequence of images using PyTorch ops based on uint8 format:
    https://github.com/tensorflow/tpu/blob/master/models/official/efficientnet/autoaugment.py#L352

Args:
        input (torch.Tensor): image tensor with shapes like :math:`(C, D, H, W)` or :math:`(B, C, D, H, W)` to equalize.

Returns:
        torch.Tensor: Sharpened image or images with same shape as the input.
"""

File: kornia/enhance/adjust.py [hunk 5]
```diff
@@ -430,46 +462,35 @@ def equalize(input: torch.Tensor) -> torch.Tensor:
...
+    input = _transform_input3d(input) * 255.
```
Comment (edgarriba) [inline kornia/enhance/adjust.py:482]: shouldn't we scale the tensor in first place inside `_scale_channel` ?

File: test/enhance/test_adjust.py [hunk 6]
```diff
@@ -599,3 +599,150 @@ def test_gradcheck(self, device):
         img = utils.tensor_to_gradcheck_var(img)  # to var
         assert gradcheck(kornia.adjust_brightness, (img, 2.),
                          raise_exception=True)
+
+
+class TestEqualize:
+    def test_shape_equalize(self, device):
```
Comment (edgarriba) [inline test/enhance/test_adjust.py:605]: add also `dtype` to tests `def test_shape_equalize(self, device, dtype):`

File: test/enhance/test_adjust.py [hunk 7]
```diff
@@ -599,3 +599,150 @@ def test_gradcheck(self, device):
...
+        bs, channels, height, width = 1, 3, 4, 5
+
+        inputs = torch.ones(channels, height, width)
```
Comment (edgarriba) [inline test/enhance/test_adjust.py:608]: pytorch recommendation to initialize tensors:
`inputs = torch.ones(channels, height, width, device=device, dtype=dtype)`

File: test/enhance/test_adjust.py [hunk 8]
```diff
@@ -599,3 +599,150 @@ def test_gradcheck(self, device):
...
+
+        channel = torch.stack([row] * height)
+        image = torch.stack([channel] * channels)
+        batch = torch.stack([image] * bs)
+
+        return batch
+
+
+class TestEqualize3D:
+    def test_shape_equalize3d(self, device):
```
Comment (edgarriba) [inline test/enhance/test_adjust.py:680]: same as above for `dtype`

**Review by edgarriba — CHANGES_REQUESTED** [review]
LGTM. Just few minor changes

**Review by edgarriba — APPROVED** [review]

**Comment by vribeiro1** [comment]
As requested by @shijianjian, I have developed the code for `RandomEqualize3D`. I will open a new PR on top of this branch so you can review. 
On this new PR, I added tests for `RandomEqualize` as well, and fixed a potential bug on it after testing.

**Comment by shijianjian** [comment]
@vribeiro1 Can you rebase? Your commits are duplicated twice here.

**Comment by vribeiro1** [comment]
@shijianjian Sorry about that. I rebased and double checked. Now it looks fine. I also solved a conflict we had with master in one docstring. If you find anything weird, please, let me know :)

File: kornia/enhance/adjust.py [hunk 9]
```diff
@@ -428,60 +429,86 @@ def _blend_one(input1: torch.Tensor, input2: torch.Tensor, factor: Union[float,
...
+        # can't index using 2d index. Have to flatten and then reshape
+        result = torch.gather(build_lut(histo, step), 0, im.flatten().long())
+        result = result.reshape_as(im)
+
+    return result / 255.
 
+
+def equalize(input: torch.Tensor) -> torch.Tensor:
+    r"""Apply equalize on the input tensor.
+
```
Comment (shijianjian) [inline kornia/enhance/adjust.py:467]:
```suggestion
```
